addons/website_event_sale/controllers/main.py

```diff
@@ -58,7 +63,12 @@ def _create_attendees_from_registration_post(self, event, registration_data):
     def registration_confirm(self, event, **post):
         res = super().registration_confirm(event, **post)
 
+
         registrations = self._process_attendees_form(event, post)
+        filtered_registration = self.registration_filter(registrations)
+        if len(filtered_registration.get('free_registration_data')) == len(registrations):
+            # skip SO creation/payment process
+            return res
```

Comment:
Hello @awa-odoo 
Actually, this is required in a use case where suppose we have some product in the cart from a previous shopping session and when we register through the ticket with `price=0` even in that case it will redirect us to the payment process instead of that confirmation page because we have something already in our cart. As no SO will be created for the free ticket so ideally we need that confirmation page instead of that payment process. Or am I missing something?

Reply:
Ah I see.

Than would simplify and do it as follows:
```py
order_sudo = request.website.sale_get_order()
if any(line.event_ticket_id for line in order_sudo.order_line):
    # order does not contain any tickets, meaning we are confirming a free event
    return res
```

Reply:
probably it should be `if not any(line.event_ticket_id for line in order_sudo.order_line):`
it will work but on the other hand quotation will be created for the product existing on the cart(I think it's ok) because it is not emptying the cart and whenever the payment process is executed all the products added to the cart will share the same sale order. or am I missing something?

If not then probably we can create `event_ticket_by_id`(in your below suggestions) in a separate method and use it on both places